### ISSUE TITLE
Fuzzy Custom Action

### ISSUE THREAD

**Issue by RickySupriyadi** [logancyang/obsidian-copilot#614]
Since this plugin will support mobile...
1. In YouTube android app, share a video to obsidian
2. obsidian will catch the link (see picture) 
![Screenshot_20240806-102856](https://github.com/user-attachments/assets/1917692e-c80b-4a22-ac65-b50d388f1014)
3. rather than only 3 catch option add 1 more, "copilot"

when choose "copilot" it will ask user to choose which "fuzzy custom action".
this custom action basically only prompt to Google gemini model to generate summary of YouTube link given.

input = {share catcher}
model = gemini 
Prompt = summary this {share catcher}
output = 
{share catcher} 
{prompt result}

**Comment by logancyang** [comment]
How's your experience using Gemini for youtube summary? Can it summarize videos without a transcript?

**Comment by RickySupriyadi** [comment]
sorry for the absurd feature request,
through gemini.google.com yes it can pull youtube links directly

but through API i never have success make it work, although gemini have 1million context length and video multimodality too... what i did is manually :
1. pull transcript (i forgot which api call i used)
2. fixed the transcript for error and make the summary (Prompt: "you are about to fix a transcript from error, mistype, and mispronounce, here is a video transcript: {video transcript} after finish fixing the transcript summary the result")

**Comment by logancyang** [comment]
@RickySupriyadi not absurd at all, I plan to add this as a Copilot command.

At one point I tried some custom GPTs and a few online tools for youtube summarization, the majority of them just pulled the ready-made transcript that is auto-generated by youtube, they don't really run through something like Whisper or a multimodal model. I plan to do the latter for videos without an auto-generated transcript.

Could you check which API you used to pull the ready-made transcript? I have a script to download youtube videos in batch and run through whisper, but that takes quite some time to run. I'll try gemini API as well to see if it's a better approach.

Just tried gemini.google.com, it can indeed work on both kinds of videos, with or without transcript. For videos without transcript it runs a bit slower but still surprisingly fast. It'll be great if I can use their API for this directly but I guess they reserve youtube exclusively for gemini cuz they are Google. Batch downloading youtube videos for my own transcribing service might get blocked?

**Comment by RickySupriyadi** [comment]
If im not mistaken i used this API https://github.com/jdepoix/youtube-transcript-api
and i got stuck with formatter something, but some how it worked out, so i didn't touch it just use it... 
I'm still looking for that folder, i hope it isn't saved in my broken laptop :/
as for formatter it is... txt, then that txt i use it for my copy paste to gpt3.5 or bard at that time, gpt 3.5 fixed the transcript more better than bard. 

nowdays i just feed gemini.Google.com for QnA of a video then copy the whole conversation into gpt4o to write a report in journalist style in markdown code block for obsidian notes format. 

btw don't try download YouTube video i heard it's against YouTube ToS or something...
but transcript i think if it's less than.... 1000 call per day... it's ok [youtube v3](https://developers.google.com/youtube/v3/guides/quota_and_compliance_audits?source=sh%2Fx%2Fsrp%2Fwr%2Fm1%2F6&kgs=30650d6ff7c4d860)

**Comment by logancyang** [comment]
Thanks for reporting this, and sorry for the slow reply.

Copilot v3 is in code freeze, so we're closing this as not planned. v4 is a ground-up rewrite and is launching soon, with a very different architecture around agent mode.

If this is still an issue for you in v4 once it lands, please open a new issue and we'll take a fresh look. 🙏
